### DISCUSSION TITLE
ValueError: A `Concatenate` layer requires inputs with matching shapes except for the concat axis

### DISCUSSION THREAD

**Discussion by amirnasab** [voxelmorph/voxelmorph#346] [Q&A]
I wanted to use voxelmorph pretrained brain_3D model and test my data 
my input data is in shape of:
(119,512,512)
and ive kept nb_features as deflaut:
nb_features = [
    [16, 32, 32, 32],
    [32, 32, 32, 32, 32, 16, 16]
]

but it buils up an error:
ValueError: A `Concatenate` layer requires inputs with matching shapes except for the concat axis. Got inputs shapes: [(None, 28, 128, 128, 32), (None, 29, 128, 128, 32)]

I cant get what the problem is. can anyone help me?

**Comment by Ehsan-data** [comment]
use this code to resize your data. Hope it  helps:

import itertools

def resize_data(data): 
    initial_size_x = data.shape[0]
    initial_size_y = data.shape[1]
    initial_size_z = data.shape[2]
    new_size_x = 160 
    new_size_y = 192 
    new_size_z = 224 
    delta_x = initial_size_x / new_size_x 
    delta_y = initial_size_y / new_size_y 
    delta_z = initial_size_z / new_size_z 
    new_data = np.zeros((new_size_x, new_size_y, new_size_z)) 
    for x, y, z in itertools.product(range(new_size_x), range(new_size_y), range(new_size_z)): 
        new_data[x][y][z] = data[int(x * delta_x)][int(y * delta_y)][int(z * delta_z)] 
    return new_data
